### PR TITLE
Approvall id code system generator

### PR DESCRIPTION
**Pull request by epuzanov**
CodeSystem ApprovallID generator class implementation.

1. ApprovalID class is configurable now.
2. hard coded AprovallID UNII alias removed from UI (can be added again to new GSRSFrontend as a config.json parameter)
3. aprovallID column length increased to feet EUTCT code as ApprovalID

Configuration Parameters:
ix.ginas.approvalIDGenerator.name - ApprovallID alias name (default: UNII)
ix.ginas.approvalIDGenerator.class - ApprovallID generator class name, is optional if class name based on AprovallID alias name (o.e: ix.ginas.utils.**UNII**Generator)

Configuration Examples:
**EUTCT CodeSystem AprovalID generator:**
ix.ginas.approvalIDGenerator = {
    "name":"EUTCT",
    "class":"ix.ginas.utils.CodeSystemGenerator"
}

### CODE REVIEW

**Comment by dkatzel-ncats** [comment]
Thanks Egor! 

I will discuss these changes with Tyler. We should figure out a solution to the approvalID column length so each agency doesn't need to change the data model

**Comment by dkatzel-ncats** [comment]
I've changed the code a bit to make it more reuseable by other agencies.  The main difference is instead of hardcoding which approval ID generator  to use it is specified in the config.  In addition I've made it so you can give additional parameters, similar to the validators, that will be passed to your generator class.  This uses Jackson's `convertValue` method passing in a Map of the parameters.  So either the object needs to have an empty constructor along with getters and setters for all the fields in the parameter map or uses a `@JsonCreator` annotated constructor or factory method.  I've gone ahead and annotated the `CodeSystemGenerator` class for you.

So to use your code in the ginas.conf you would add
```
ix.ginas.approvalIDGenerator = {
 "generatorClass" : "ix.ginas.utils.CodeSystemGenerator",
 "parameters" : {
                 "name" : "EUTCT",
                 "codeSystem" : "EUTCT"
                }

}
```

After talking with Niko and Tyler we decided by default we don't want people making UNIIs so to avoid confusion I added a new generator called DefaultApprovalIDGenerator which is very similar to the UNII but is a different number of characters and has an optional prefix.

I also added a `getName()` method to the approval Generators so we can get the name and put it in the UI.  I've updated the legacy UI but for the new beta UI you will have to modify it's config separately to match.  We thought about making a new API request to get the name but didn't want to add more network chatter for every request.

File: modules/ginas/app/ix/ginas/utils/UniiLikeGenerator.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    public static char[] alphabet = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ".toCharArray();
+    //GSRS-1658 can not contain only digit characters and "E" characters (`^[0-9E]*$`)
+    private static Pattern SCI_NOTATION_PATTERN = Pattern.compile("^[0-9E]+$");
+    //GSRS-1658 can not contain month abbreviations which cause problems in excel
+    private static Set<String> MONTHS = new HashSet<>(Arrays.asList("JAN", "FEB", "MAR", "APR", "MAY", "JUN", "JUL", "AUG", "SEP", "OCT", "NOV", "DEC"));
+    private static List<Character> CONFUSING_CHARS_LIST = Arrays.asList('0', 'O', '1', 'I');
+
+    private static char[] CONFUSING_CHARS_ARRAY;
+    static{
+        CONFUSING_CHARS_ARRAY = new char[CONFUSING_CHARS_LIST.size()];
```
Comment (tylerperyea) [inline modules/ginas/app/ix/ginas/utils/UniiLikeGenerator.java:21]: I have an aversion to static blocks. I may have rewritten this as a cachedSupplier, but it's a matter of preference.

**Review by tylerperyea — APPROVED** [review]
Awesome work guys! Looks good!

**Pull request merged**